src/structures/AutocompleteInteraction.js

```diff
@@ -0,0 +1,114 @@
+'use strict';
+
+const CommandInteractionOptionResolver = require('./CommandInteractionOptionResolver');
+const Interaction = require('./Interaction');
+const { InteractionResponseTypes, ApplicationCommandOptionTypes } = require('../util/Constants');
+
+/**
+ * Represents an autocomplete interaction.
+ * @extends {Interaction}
+ */
+class AutocompleteInteraction extends Interaction {
+  constructor(client, data) {
+    super(client, data);
+
+    /**
+     * The channel this interaction was sent in
+     * @type {?TextBasedChannels}
+     * @name AutocompleteInteraction#channel
+     * @readonly
+     */
+
+    /**
+     * The id of the channel this interaction was sent in
+     * @type {Snowflake}
+     * @name AutocompleteInteraction#channelId
+     */
```

Comment:
Is this actually needed?

Reply:
I think so, `AutocompleteInteraction` always has a channelId, which is not guaranteed in `Interaction`.